core/src/com/unciv/logic/automation/UnitAutomation.kt

```diff
@@ -188,14 +214,26 @@ object UnitAutomation {
 
         var bestTilesForHealing = tilesByHealingRate.maxBy { it.key }!!.value
         // within the tiles with best healing rate (say 15), we'll prefer one which has the highest defensive bonuses
-        val bestTilesWithoutBombardableTiles = bestTilesForHealing.filterNot { it.getTilesInDistance(2)
-                .any { it.isCityCenter() && it.getOwner()!!.isAtWarWith(unit.civInfo) } }
-        if(bestTilesWithoutBombardableTiles.any()) bestTilesForHealing = bestTilesWithoutBombardableTiles
-        val bestTileForHealing = bestTilesForHealing.maxBy { it.getDefensiveBonus() }!!
+        //val bestTilesWithoutBombardableTiles = bestTilesForHealing.filterNot { it.getTilesInDistance(3)
+        //        .any { it.isCityCenter() && it.getOwner()!!.isAtWarWith(unit.civInfo) } }
+        //if(bestTilesWithoutBombardableTiles.any()) bestTilesForHealing = bestTilesWithoutBombardableTiles
+
+        // Dirty Hack
+        bestTilesForHealing = (((bestTilesForHealing subtract tilesInRangeOfAttack) union (listOf(currentUnitTile))).toList())
```

Comment:
Here it's slightly more complicated since we're subtracting, probably bestTilesForHealing should be converted to a hashmap